src/mvc/collection.js

```diff
@@ -0,0 +1,109 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+'use strict';
+
+/**
+ * Collections are ordered sets of models.
+ *
+ * Any event that is triggered on a model in a collection will also be triggered on the collection directly, for
+ * convenience.
+ *
+ * @class Collection
+ */
+
+CKEDITOR.define( [ 'emitter', 'utils' ], function( EmitterMixin, utils ) {
+	function Collection() {
+		/**
+		 * The internal list of models in the collection.
+		 *
+		 * @property _models
+		 * @private
+		 */
+		Object.defineProperty( this, '_models', {
+			value: []
+		} );
+
+		/**
+		 * The number of items available in the collection.
+		 *
+		 * @property length
+		 */
+		Object.defineProperty( this, 'length', {
+			get: function() {
+				return this._models.length;
+			}
+		} );
+	}
+
+	/**
+	 * @inheritdoc utils#extend
+	 */
+	Collection.extend = utils.extendMixin;
+
+	utils.extend( Collection.prototype, EmitterMixin, {
+		/**
+		 * Adds an item into the collection.
+		 *
+		 * Note that this is an array-like collection, so the same item can be present more than once. This behavior is
+		 * for performance purposes only and is not guaranteed to be kept in the same way in the future.
+		 *
+		 * @param {Model} model The item to be added.
+		 */
+		add: function( model ) {
+			model.addParentEmitter( this );
```

Comment:
Looking purely to the model side of our MVC api, I don't see a big deal because I don't expect models to fire events other than `change`.

In the other hand, we're talking about the events API, which will not be used exclusively for model purposes but for everything that needs events.

Therefore, if we decide NOT to [go with general event bubbling](https://github.com/ckeditor/ckeditor5-design/issues/7#issuecomment-73016905), then I agree that we can remove the `addParentEmitter()` method, handling bubbling in the way you proposed, exclusively inside Collection and for the `change` event only.